src/omlt/base/constraint.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+from typing import Any
+
+import pyomo.environ as pyo
+from pyomo.core.expr import EqualityExpression, InequalityExpression
+
+from omlt.base import DEFAULT_MODELING_LANGUAGE
+
+
+class OmltConstraint:
+    @property
+    def ctype(self):
+        return pyo.Constraint
+
+    def valid_model_component(self):
+        """Return True if this can be used as a model component."""
+        return True
+
+
+class OmltConstraintScalar(OmltConstraint):
+    def __init__(self, lang: str = DEFAULT_MODELING_LANGUAGE, **kwargs: Any):
+        lhs = kwargs.pop("lhs", None)
+        if lhs is not None:
+            self.lhs = lhs
+        sense = kwargs.pop("sense", None)
+        if sense is not None:
+            self.sense = sense
+        rhs = kwargs.pop("rhs", None)
+        if rhs is not None:
+            self.rhs = rhs
+        if not lhs and not sense and not rhs:
+            expr = kwargs.pop("expr", None)
+            if isinstance(expr, EqualityExpression):
+                self.lhs = expr.arg(0)
+                self.sense = "=="
+                self.rhs = expr.arg(1)
+            if isinstance(expr, InequalityExpression):
+                self.lhs = expr.arg(0)
+                self.sense = "<="
+                self.rhs = expr.arg(1)
```

Comment:
it might lead to confusing behavior if it's possible to provide 'lhs', 'rhs', 'sense' and 'expr' in kwargs. in this case, 'expr' would be ignored. what do you think?